src/main/java/marquez/core/mappers/Mapper.java

```diff
@@ -1,7 +1,17 @@
 package marquez.core.mappers;
 
+import static java.util.Objects.requireNonNull;
+
 import java.util.Optional;
 
-interface Mapper<A, B> {
-  Optional<B> map(A value);
+public abstract class Mapper<A, B> {
+  public Optional<B> mapAsOptional(A value) {
+    return Optional.ofNullable(map(value));
+  }
+
+  public Optional<B> mapIfPresent(Optional<A> value) {
+    return requireNonNull(value).map(this::map);
+  }
```

Comment:
Could these both just be called `map`  with overloading?

```
  public Optional<B> map(A value) {
...
   public Optional<B> map(Optional<A> value) {
```

Reply:
I tend to favor descriptive method names over overloading methods with the same set of parameter. But, it would simplify the mapper contract, so sure!

Reply:
OK, no reason to change it just because of my comment -- it's your call. Just wanted to make sure I understood the alternative.

Reply:
Well, for sure `Mapper. mapIfPresent()` if I'm given the choice :)